packages/covector/src/run.ts

```diff
@@ -219,8 +223,154 @@ export function* covector({
 
     return pkgCommandsRan;
   } else if (command === "preview") {
-    console.log('WIP');
-    return 'placeholder';
+    yield raceTime({ t: config.timeout });
+
+    let versionTemplate;
+    
+    switch(previewVersion){
```

Comment:
I think for the `versionTemplate`, we need to allow someone to configure an "input" value. This might be a branch name or a `-next.N`. Then the `N` could be the `date` in this case.

I think that means the possible output could be:
- `x.x.x-{identifier}.{timestamp}` where `indentifier` could be "next" or {branch_name}.
- `x.x.x-{timestamp}`
- `x.x.x-{identifier}.{sha}`

I don't think we want to allow them to use `x.x.x-{sha}` because of the resolution issues that cause it to sort in an unexpected order. Even with a `.{sha}` I suspect that it is possible to cause issues.

I did use `prereleaseIdentifier` in my PR, but it may not actually overlap at all?

Reply:
i can add an identifier input option or we can have the branch name just be the hard coded default.

> I don't think we want to allow them to use x.x.x-{sha} because of the resolution issues that cause it to sort in an unexpected order. Even with a .{sha} I suspect that it is possible to cause issues.

No matter how we format it, SHA will mess with the sorting order. But also, running `yarn update` is flakey from our findings. So did you want me to just scrap sha altogether?

Reply:
> i can add an identifier input option or we can have the branch name just be the hard coded default.

This sounds good to me. Input for cases like `-next` and hard-coded otherwise. Maybe double check with what I did in my PR so we don't step on each other's toes regarding the identifier (but I don't think we will?).

> So did you want me to just scrap sha altogether?

The `yarn update` issue is pretty bad. I keep thinking of your comment, "it is nice to see that I installed it with the sha", and how we can enable that without messing up the sort order. Any other ideas? Do we bother with the `+{buildmetadata}` and put it there?

Reply:
Could you remind me what sort order you're talking about? And what's buildmetadata?

Reply:
The branch names in my preview action would convert dashes to underscores and underscores to two underscores. I don't remember the reasoning behind it but I assume it'll help with semver because it uses dash for its prerelease pattern? Should i apply the same conversion for the branch identifier?

Reply:
The sort order would be if someone specifies `^1.2.9-beta.5`, which version will they actually receive? It will give you the tip of the series, and that tip is determined by the sort order.

I think that might be because only `-` is valid? From [semver.org](https://semver.org/#backusnaur-form-grammar-for-valid-semver-versions).

```
<non-digit> ::= <letter>
              | "-"
```

Reply:
👍 I added a `identifier` argument. The default value of it is `branch` and it's the only option at the moment (so we can add more in the future) so it doesn't need to be specified in the workflow.

> I think that might be because only - is valid? From semver.org.

I confused versioning with tags. 

I transform the branch name so that `_` and `/` becomes `-`.

![Screen Shot 2021-04-27 at 5 14 48 PM](https://user-images.githubusercontent.com/29791650/116314918-be46e580-a77d-11eb-97d5-3ec26b7e221c.png)